### PR TITLE
Fix extractData to work with non float cols and with params

### PR DESCRIPTION
**Pull request by smartsammler**
- Add possibility to use extractData for columns with e.g. strings
  instead of floats.
- Use precision used by the SDDS file.
- Add possibility to ask for parameters and not just columns.
- Change `getAllCols` to use `sdds`
- Add `getAllPars` similar to `getAllCols`
- _New dependency_: **sdds** (the official sdds Python wrapper)
  - http://www.aps.anl.gov/Accelerator_Systems_Division/Accelerator_Operations_Physics/software.shtml#PythonBinaries
  - http://www.aps.anl.gov/Accelerator_Systems_Division/Accelerator_Operations_Physics/installationGuide_Linux.shtml
  - Unfortunately there is no Package in the PyPI for SDDS at the moment. Furthermore the binaries are for Python 2 only though it is possible to compile the important file for Python 3.

### CODE REVIEW

File: beamline/datautils.py [hunk 1]
```diff
@@ -55,24 +59,80 @@ def getAllCols(self, sddsfile=None):
...
-            self.dcmdline += ' -col={kw},format={p}'.format(kw=k, p=self.precision)
-        cmdlist = ['bash', self.dscript, self.dpath, self.dcmdline]
-        retlist = []
-        proc = subprocess.Popen(cmdlist, stdout=subprocess.PIPE)
-        for line in proc.stdout:
-            retlist.append([float(i) for i in line.split()])
-        self.h5data = np.array(retlist)
+        columns = self.sddsobj.columnName
+        parameters = self.sddsobj.parameterName
+        data = [self.sddsobj.columnData[columns.index(col)][0]
```
Comment (zhangt58) [inline beamline/datautils.py]: In my laptop, `sddsobj.columnData` just gives a list of `[None]`, `sddsobj.parameterData` works right, so I cannot verify the test script here. Maybe it it the issue of the SDDS python interface.

File: setup.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ def readme():
     with open('README.rst') as f:
         return f.read()
 
-requiredpackages = ['pyrpn', 'h5py', 'numpy', 'matplotlib', 'pyepics']
+requiredpackages = ['pyrpn', 'h5py', 'numpy', 'matplotlib', 'pyepics', 'sdds']
```
Comment (zhangt58) [inline setup.py]: Could you please just make `sdds` optional, for the compatibility?

**Review by zhangt58 — CHANGES_REQUESTED** [review]
@smartsammler Good job! In order to be compatible with the already deployed `beamline` version, it is better to provide the pure python SDDS interface as an extra option, along side the previous external SDDS commands, e.g. the following should be done:
- handle import exceptions when `sdds` cannot be imported, in this case, rollback to the external callback approach;
- Can you make the `sdds` python package, `.whl` or `.egg`? I built it from source code, and could work on my laptop, however some method does not give expected results.

If you including both the pure python and external approaches, I will be glad to merge, thanks!

**Comment by smartsammler** [comment]
Thank you for your input and sorry for the delay.
Should I make a new commit or should I squash/amend the new commit with this PR? (I'm sorry I have no experience with PR reviews so far)

May you test the wheel, before I try to publish it? Unfortunately I needed to zip it to upload it to GitHub
[sddswhl.whl.zip](https://github.com/archman/beamline/files/539658/sddswhl.whl.zip)

**Comment by zhangt58** [comment]
I've tested with your `sdds .whl`, this time `sddsobj.columnData` gives right results, that's great!
You can just continue to commit to the PR branch when it is still open (i.e. not merged), I think the new commits will automatically show up.
Thank you!

**Pull request merged**